torchtune/models/qwen2/_model_builders.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
```

Comment:
Is there generally interest in the 1.5B model too? If so, is it worth adding?

Perhaps as a more general point, I noticed in the Qwen2 technical report that weight tying is only used for the 0.5B and 1.5B models, right? Would users want to configure weight tying in the larger models where weight tying is enabled by default?

Reply:
Yes, I will add other size of models (0.5B / 1.5B / 72B).
In addition, whether weight tying is enabled must be consistent between model initialization (in `_model_builders.py` here) and weight loading (in `_convert_weights.py`, reading configuration from `config.json` file), so I think we should fix the weight tying configuration in default model providers to avoid users from bugs.
Based on the experience of the open source community, most users have no need to modify weight tying when tuning Qwen models. For expert users, they can directly use `torchtune.models.qwen2.qwen2()` to provide correct hyperparameters.

Reply:
Thanks for the insight!

This brings me to a broader point and feel free to ignore me if this sounds crazy. If we don't think there would be interest from the community in the 0.5B/1.5B models, which do use weight tying, would we be able to simplify this PR a bunch by just using the original `TransformerDecoder` for the 7B and 72B models, which don't need weight tying? 

If you think it's worth offering this flexibility I think the added decoder is fine, and it'll be nice to share it with Gemma. (also cc @joecummings @ebsmothers here).

Reply:
Yeah, we've made it pretty far in this PR so I think best course of action is to land this with the weight tying; however, for future reference @SalmanMohammadi is absolutely correct that we prefer to start by landing *only* what is needed, which - in this case - would just be regular Qwen2 without weight tying.

This code will make it easy to add the small models, which is a huge benefit for our memory constrained users, so I'm excited for the quick follow up with those configs :)